Source/CombatExtended/Harmony/Harmony_GenRadial.cs

```diff
@@ -79,54 +79,92 @@ internal static void Patch()
 
         public static bool NumCellsInRadius(out int __result, float radius)
         {
-            if (radius < 1f) // special case since we start scanning from the *previous* index, which would be negative in this instance.
+            // Special cases
+            if (radius < 2f || radius >= MAX_RADIUS)
             {
-                __result = 1;
-                return false;
-            }
-            var radialPatternNumCells = RadialPatternNumCells; // cache friendly local references
-            var radialPatternRadii = RadialPatternRadii;
-            if (radius >= MAX_RADIUS)
-            {
-                if (radius > MAX_RADIUS)
+                if (radius < 0f)
+                    __result = 0;
+                else if (radius < 1f)
+                    __result = 1;
+                else if (radius < 1.414213562f)
+                    __result = 5;
+                else if (radius < 2f)
+                    __result = 9;
+                else
                 {
-                    Log.Error($"Requested radius {radius} is beyond max. Truncating to {Harmony_GenRadial.MAX_RADIUS}.");
+                    __result = RadialPatternCount;
+                    CheckRadius(radius);
                 }
-                __result = RadialPatternCount;
                 return false;
             }
-            float radsq = radius * radius;
-            int r = (int)radsq;
-            int count = radialPatternNumCells[r - 1];
-            /*
-              If we raise the max radius above about 200, binary search becomes faster.
-              Below 200, the match will reliably be found within 64 tries, where linear memory access dominates.
-            int max_count = RadialPatternNumCells[r];
-            int span;
-            */
+
+            int radiusSquaredInt = (int)(radius * radius);
+            // The answer is at least (25/8 * radius * radius - 11)
+            int lowerBound = ((radiusSquaredInt * 25) >> 3) - 11;
 #if BINSEARCH
-            while ((span = max_count - count) > 64)
+            // The upper bound is set to (101/32 * radius * radius)
+            //   some small radius actually goes above that
+            //   but the error is small (< 16) and linear search can fix it
+            int upperBound = (radiusSquaredInt * 101) >> 5;
```

Comment:
When constants aren't obvious in their origin, it's best to name them. At the least, we need a comment either explaining how you arrive at it or a link to an article explaining.

Reply:
I found those constants (this one as well as the 25/8) myself, but turns out simply (pi * r * r) is more efficient if I just implement linear searching in both directions.